src/transport/raw/MessageHeader.cpp

```diff
@@ -200,19 +197,13 @@ CHIP_ERROR PayloadHeader::Decode(Header::Flags flags, const uint8_t * const data
     uint8_t header;
     uint16_t octets_read;
 
-    err = reader.Read8(&header);
-    SuccessOrExit(err);
-
-    err = reader.Read8(&mMessageType);
-    SuccessOrExit(err);
-
-    err = reader.Read16(&mExchangeID);
+    err = reader.Read8(&header).Read8(&mMessageType).Read16(&mExchangeID).StatusCode();
     SuccessOrExit(err);
 
     if (flags.Has(Header::FlagValues::kVendorIdPresent))
     {
         uint16_t vendor_id;
-        err = reader.Read16(&vendor_id);
+        err = reader.Read16(&vendor_id).StatusCode();
```

Comment:
why don't we read the vendor id directly? is it the wrong size/type? should we instead make the type match?

Reply:
`mVendorId` is an `Optional<uint16_t>`.  So yes,it's the wrong type for reading directly, but it's the right type for what we are trying to express.

We _could_ address this by adding a method on `Optional` that sets it into the "have value" state and returns a reference to the value to be filled in.  Then we could write something like:
```
  err = reader.Read16(&mVendoId.SetValue()).StatusCode();
```
or something.  I wasn't sure whether it was desirable to add such an API to `Optional`; I've seen it on `Optional`-like things in the past, but there are safety tradeoffs with having an API that temporarily leaves the object in an invalid state...

Reply:
Ah, forgot about optional. Existing code makes sense then.